In [15]:
# %matplotlib widget

from __future__ import annotations


import matplotlib.colors as mcolors
from test_utilities import test, TestDict

COLORS = list(mcolors.CSS4_COLORS.keys())

<link href="style.css" rel="stylesheet"></link>
<article class="day-desc"><h2>--- DPart I ---</h2></article>


In [ ]:
tests: list[TestDict] = [
    {
        "name": "Example",
        "s": """
            162,817,812
            57,618,57
            906,360,560
            592,479,940
            352,342,300
            466,668,158
            542,29,236
            431,825,988
            739,650,466
            52,470,668
            216,146,977
            819,987,18
            117,168,530
            805,96,715
            346,949,466
            970,615,88
            941,993,340
            862,61,35
            984,92,344
            425,690,689
        """,
        "n": 10,
        "expected": 40,
    },
]


@test(tests=tests)
def part_I(s: str, n: int) -> int:
    return -1


Test Example passed, for part_I.
Success


In [ ]:
with open("../input/day10.txt") as f:
    puzzle = f.read()

# print(f"Part I: {part_I(puzzle, 1000)}")

Part I: 52668


<link href="style.css" rel="stylesheet"></link>
<main>

<p>Your puzzle answer was <code>52668</code>.</p><p class="day-success">The first half of this puzzle is complete! It provides one gold star: *</p>


<link href="style.css" rel="stylesheet"></link>
<article class="day-desc"><h2 id="part2">--- Part Two ---</h2></article>



Test Example passed, for part_II.
Success


<link href="style.css" rel="stylesheet"></link>


In [ ]:
# print(f"Part II: {part_II(puzzle)}")

Part II: 1474050600


<link href="style.css" rel="stylesheet"></link>
<main>

<p>Your puzzle answer was <code>1474050600</code>.</p><p class="day-success">Both parts of this puzzle are complete! They provide two gold stars: **</p>

</main>
